File: data/EXP01/ruby/1/1.output_2.rb
def two_sum(nums, target)
    num_hash = {}
    
    nums.each_with_index do |num, index|
        complement = target - num
        if num_hash.include?(complement)
            return [num_hash[complement], index]
        end
        num_hash[num] = index
    end
    
    return []
end

File: data/EXP01/csharp/1/1.output_2.go
func twoSum(nums []int, target int) []int {
    compMap := make(map[int]int)
    
    for i, num := range nums {
        
        if j, ok := compMap[num]; ok {
            return []int{j, i}
        }
        
        compMap[target-num] = i
    }
    
    return []int{}
}
